types/system-task/index.d.ts

```diff
@@ -2,25 +2,59 @@
 // Project: https://github.com/leocwlam/system-task
 // Definitions by: Leo Lam <https://github.com/leocwlam>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.1
 
-/// <reference types="node" />
+declare class SystemTask {
+    type: string;
 
-export = SystemTask;
+    constructor(taskType?: string, isAsyncProcess?: boolean, logMethod?: any);
 
-declare class SystemTask {
-  type: string;
-  constructor(taskType: string, isAsyncProcess?: boolean, logMethod?: any);
-  log(type: string, message: string, detail?: any): void;
-  insertPreprocessItemsHandler(task: SystemTask): Promise<any>;
-  preprocessHandler(task: SystemTask, preProcessItem: any): Promise<any>;
-  processHandler(task: SystemTask, processItem: any): Promise<any>;
-  cleanupHandler(task: SystemTask, cleanupItems: any[]): Promise<any>;
-  isValidProcess(): void;
-  start(): void;
+    /**
+     * @async
+     */
+    log(type: string, message: string, detail?: any): void;
```

Comment:
Async, but not promise-returning?

Reply:
that's OK IMO, formally JSDoc will sign any of those method as `async` (for async/await). We can tag those methods `@async`, as those are expected to be used as (via readme):
```ts
async processHandler (task, processItem) {
    await task.log('info', `execute`, {Type: task.type, processItem})
    // processItem can be execute method 
    // e.g. await processItem.execute()

    return processItem
  }
```
https://jsdoc.app/tags-async.html

Reply:
Thanks for checking!